crates/bevy_time/src/fixed.rs

```diff
@@ -0,0 +1,345 @@
+/// the de-facto performance target for many games).
+///
+/// To run a system on a fixed timestep, add it to the
+/// [`FixedUpdate`](bevy_app::FixedUpdate) schedule. This schedule is run a
+/// number of times between [`PreUpdate`](bevy_app::PreUpdate) and
+/// [`Update`](bevy_app::Update) according to the accumulated
+/// [`overstep()`](Time::overstep) time divided by the
+/// [`timestep()`](Time::timestep). This means the schedule may run 0, 1 or more
+/// times during a single update (which typically corresponds to a rendered
+/// frame).
+///
+/// `Time<Fixed>` and the generic [`Time`] resource will report a
+/// [`delta()`](Time::delta) equal to [`timestep()`](Time::timestep) and always
+/// grow [`elapsed()`](Time::elapsed) by one [`timestep()`](Time::timestep) per
+/// iteration.
+///
+/// The fixed timestep clock follows the [`Time<Virtual>`](Virtual) clock, which
+/// means it is affected by [`pause()`](Time::pause),
+/// [`set_relative_speed()`](Time::set_relative_speed) and
+/// [`set_max_delta()`](Time::set_max_delta) from virtual time. If the virtual
+/// clock is paused, the [`FixedUpdate`](bevy_app::FixedUpdate) schedule will
+/// not run. It is guaranteed that the [`elapsed()`](Time::elapsed) time in
+/// `Time<Fixed>` is always between the previous `elapsed()` and the next
+/// `elapsed()` value in `Time<Virtual>`, so the values are compatible.
+///
+/// Changing the timestep size while the game is running should not normally be
+/// done, as having a regular interval is the point of this schedule, but it may
+/// be necessary for effects like "bullet-time" if the normal granularity of the
+/// fixed timestep is too big for the slowed down time. In this case,
+/// [`set_timestep()`](Time::set_timestep) and be called to set a new value. The
+/// new value will be used immediately for the next run of the
+/// [`FixedUpdate`](bevy_app::FixedUpdate) schedule, meaning that it will affect
+/// the [`delta()`](Time::delta) value for the very next
+/// [`FixedUpdate`](bevy_app::FixedUpdate), even if it is still during the same
+/// frame. Any [`overstep()`](Time::overstep) present in the accumulator will be
+/// processed according to the new [`timestep()`](Time::timestep) value.
+#[derive(Debug, Copy, Clone, Reflect)]
+pub struct Fixed {
+    timestep: Duration,
+    overstep: Duration,
+}
+
+impl Time<Fixed> {
+    /// Corresponds to 64 Hz.
+    const DEFAULT_TIMESTEP: Duration = Duration::from_micros(15625);
+
+    /// Return new fixed time clock with given timestep as
+    /// [`Duration`](std::time::Duration)
+    ///
+    /// # Panics
+    ///
+    /// Panics if `timestep` is zero.
+    pub fn from_duration(timestep: Duration) -> Self {
+        let mut ret = Self::default();
+        ret.set_timestep(timestep);
+        ret
+    }
+
+    /// Return new fixed time clock with given timestep seconds as `f64`
+    ///
+    /// # Panics
+    ///
+    /// Panics if `seconds` is negative or not finite.
```

Comment:
Good catch. The asserts and the documentation needs to be corrected. It would have still panicked on zero though, see [`set_timestep`][1].

[1]: https://github.com/bevyengine/bevy/blob/95234cfb401fd5875a4cc67c5e204b76278c44e8/crates/bevy_time/src/fixed.rs#L123

Reply:
Thanks for spotting, fixed the docstring, also for `set_timestep_seconds`.